DABI.js

```diff
@@ -0,0 +1,677 @@
+FW.doWeb = function(c, a) {
+    var b = FW.getScraper(c, a);
+    b.makeItems(c, a, [], function(f, e, g, d) {
+        e.callHook("scraperDone", f, g, d);
+        if (!f.title) {
+            f.title = ""
+        }
+        f.complete()
+    }, function() {
+        Zotero.done()
+    });
+    Zotero.wait()
+};
+
+function detectWeb(doc, url) {
+    //namespace code
+    var namespace = doc.documentElement.namespaceURI;
+    var nsResolver = namespace ? function(prefix) {
+        if (prefix == 'x') return namespace;
+        else return null;
+    } : null;
+
+    if (doc.title.match("DABI. Datensatz Vollanzeige")) {
+        return "journalArticle";
+    } else if (doc.title.match("DABI: Rechercheergebnis")) {
+        var treffer = doc.evaluate('//*[text()="Keine Treffer für die Suche nach:"]', doc, nsResolver, XPathResult.ANY_TYPE, null).iterateNext();
+        if (treffer == null) {
+            return "multiple";
+        }
+    }
+}
+
+function doWeb(doc, url) {
+    //namespace code
+    var namespace = doc.documentElement.namespaceURI;
+    var nsResolver = namespace ? function(prefix) {
+        if (prefix == 'x') return namespace;
+        else return null;
+    } : null;
+
+    //variable declarations
+    var articles = new Array();
+    var items = new Object();
+    var nextLink;
+
+    //If Statement checks if page is a Search Result, then saves requested Items
+    if (detectWeb(doc, url) == "multiple") {
+        var links = doc.evaluate('//tr/td[1]/a', doc, nsResolver, XPathResult.ANY_TYPE, null);
+        var title = doc.evaluate('//tr/td[3]', doc, nsResolver, XPathResult.ANY_TYPE, null);
+
+        while (nextLink = links.iterateNext()) {
+            items[nextLink.href] = title.iterateNext().textContent;
+
+        }
+        items = Zotero.selectItems(items);
+        for (var i in items) {
+            articles.push(i);
+        }
+
+    } else if (detectWeb(doc, url) == "journalArticle") {
+        //saves single page items
+        articles = [url];
+    } else {
+        return false;
+    }
+
+    //Tells Zotero to process everything. Calls the "scrape" function to do the dirty work.
+    Zotero.Utilities.processDocuments(articles, scrape, function() {
+        Zotero.done();
+    });
+    Zotero.wait();
+    //Translator is FINISHED after running this line. Note: code doesn't run from top to bottom only.
+}
+
+//The function used to save well formatted data to Zotero
+function associateData(newItem, items, field, zoteroField) {
+    if (items[field]) {
+        newItem[zoteroField] = items[field];
+    }
+}
+
+function scrape(doc, url) {
+    //namespace code
+    var namespace = doc.documentElement.namespaceURI;
+    var nsResolver = namespace ? function(prefix) {
+        if (prefix == 'x') return namespace;
+        else return null;
+    } : null;
+
+    //variable declarations
+    var newItem = new Zotero.Item('journalArticle');
+    newItem.url = doc.location.href;
+    newItem.title = "No Title Found";
```

Comment:
If you have a good reason to believe that a title might be missing from metadata (i.e. you can find examples where that is the case), you should not be assigning anything to the title. Zotero automatically rejects items without titles during import and sends an error report to zotero.org. Missing titles are generally a good indicator that something about the translator is not working (probably the website was updated). This allows us to catch issues quickly.

If you do expect titles to be missing, then you should use something like `"[No Title]"` (@adam3smith do you remember if we had anything specific)